### ISSUE TITLE
Completions should be suppressed in all locations that are unambiguously definitions

### ISSUE THREAD

**Issue by pyscripter** [astral-sh/ty#1287]
### Summary

The following is based on my experience with using the ty playground. (^ is the cursor position)

In
```python
import numpy as ^
```
I get a list of modules on the python path.  Instead I should be getting an empty list as for example with Jedi.

**Comment by AlexWaygood** [comment]
By same principle:

- `for x in range(42)` creates a definition for the `x` symbol, so there shouldn't be any completion suggestions at the point `for x<CURSOR>`. Currently there are. 
- Typing a parameter name in a function declaration (`def f(xx<CURSOR>)`) should not trigger any autocomplete suggestions
- Typing a type parameter name in a function declaration (`def f[T<CURSOR>]`) should not trigger any autocomplete suggestions
- `with <EXPR> as b<CURSOR>` should not trigger any autocomplete suggestions
- `case <PATTERN> as b<CURSOR>` should not trigger any autocomplete suggestions
- `except <EXPRESSION> as b<CURSOR>` should not trigger any autocomplete suggestions

<details>
<summary>Screenshots</summary>

<img width="1138" height="664" alt="Image" src="https://github.com/user-attachments/assets/c58f4260-5688-47b2-aaf5-b15bc0bd9d6c" />

<img width="1340" height="842" alt="Image" src="https://github.com/user-attachments/assets/f140ec37-2e4f-4e20-be6e-f297713f6c64" />

<img width="1278" height="310" alt="Image" src="https://github.com/user-attachments/assets/4a11be21-8b8f-47cd-9bdc-933cc0ee07dc" />

<img width="1574" height="354" alt="Image" src="https://github.com/user-attachments/assets/8d26726d-88f6-4773-95ae-de79c9814cc0" />

<img width="1202" height="682" alt="Image" src="https://github.com/user-attachments/assets/a01d5a42-9b2f-46a7-a0ce-7c690dc02a72" />

<img width="1552" height="724" alt="Image" src="https://github.com/user-attachments/assets/0d44815b-23cf-4823-8f7a-a81d926ea8e2" />

</details>

**Comment by pyscripter** [comment]
Also  

import numpy ^  (offer as as the only option)
for x ^  (offer in as the only option)

**Comment by pyscripter** [comment]
Nice but it does not cover all cases discussed here:

- import numpy ^  (offer as as the only option)
-  for ^ (offer no definition)
- for x ^  (offer in as the only option) 
- Typing a parameter name in a function declaration (def f(xx<CURSOR>)) should not trigger any autocomplete suggestions
- Typing a type parameter name in a function declaration (def f[T<CURSOR>]) should not trigger any autocomplete suggestions

Shouldn't 

```rust
fn is_in_definition_place(db: &dyn Db, file: File, tokens: &[Token], typed: Option<&str>) -> bool {
    fn is_definition_token(token: &Token) -> bool {
        matches!(
            token.kind(),
            TokenKind::Def | TokenKind::Class | TokenKind::Type | TokenKind::As
        )
    }
```

include TokenKind::For?

**Comment by BurntSushi** [comment]
I opened https://github.com/astral-sh/ty/issues/1563 to track those.

Context aware keyword suggestions are a separate thing. I opened https://github.com/astral-sh/ty/issues/1568 to track that.

**Comment by pyscripter** [comment]
But for would fit nicely in the this commit.
